packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -266,7 +290,40 @@ class MediaQueryData {
   ///
   ///  * [MediaQuery.textScaleFactorOf], a method to find and depend on the
   ///    textScaleFactor defined for a [BuildContext].
-  final double textScaleFactor;
+  @Deprecated(
+    'Use textScaler instead. '
+    'Use of textScaleFactor was deprecated in preparation for the upcoming nonlinear text scaling support. '
+    'This feature was deprecated after v3.12.0-2.0.pre.',
+  )
+  double get textScaleFactor => textScaler.textScaleFactor;
+  final double _textScaleFactor;
```

Comment:
Let's leave a comment here explaining why `_textScaleFactor` continues to exist even though it is not wired up to the getter above.